### PR TITLE
Remove history mixin.

### PR DESCRIPTION
**Pull request by simoneduca**
Also remove annoying warning `browser.js?030c:49 Warning: [react-router] props.history and context.history are deprecated. Please use context.router.`

### CODE REVIEW

File: app/pages/projects.cjsx [hunk 1]
```diff
@@ -38,7 +35,7 @@ ProjectsPage = React.createClass
         delete query[key]
     newLocation = Object.assign {}, @props.location, {query}
     newLocation.search = ""
-    @props.history.replace newLocation
```
Comment (eatyourgreens) [inline app/pages/projects.cjsx]: Should `history` be removed wherever it's being passed in as a prop too?

Comment (rogerhutchings) [inline app/pages/projects.cjsx]: I think it should. It's passed in only for route-level components, but it's still deprecated.

Comment (rogerhutchings) [inline app/pages/projects.cjsx]: The only other thing on this - we're replacing `replace` with `push`, but I prefer it that way anyway.

Comment (simoneduca) [inline app/pages/projects.cjsx]: What Rog said :)

**Pull request merged**